src/lib/windows-helpers.ts

```diff
@@ -0,0 +1,96 @@
+import { app, dialog, shell } from 'electron';
+import path from 'path';
+import { __ } from '@wordpress/i18n';
+import sudo from 'sudo-prompt';
+import { STUDIO_DOCS_WINDOWS_SPEED_UP_SITES_URL } from '../constants';
+import { loadUserData, saveUserData } from '../storage/user-data';
+
+export async function promptWindowsSpeedUpSites( {
+	skipIfAlreadyPrompted,
+}: {
+	skipIfAlreadyPrompted: boolean;
+} ) {
+	const userData = await loadUserData();
+
+	if (
+		process.platform !== 'win32' ||
+		( skipIfAlreadyPrompted && typeof userData.promptWindowsSpeedUpResult !== 'undefined' )
+	) {
+		return;
+	}
+
+	const NOT_INTERESTED = __( "I'm not interested." );
+	const MANUAL_UPDATE = __( "I'll do it my own by following the documentation." );
+	const AUTOMATIC_UPDATE = __( 'Sounds good, do it for me.' );
+
+	const buttons = [ NOT_INTERESTED, MANUAL_UPDATE, AUTOMATIC_UPDATE ];
```

Comment:
When looking at the dialog, I think it would feel more intuitive for me to see the `Sounds good, do it for me` option at the top of the list, not the bottom. 

I also wondered if it'd make sense for the `I'm not interested` option to follow the same `Don't ask again` pattern we use for the demo update and arm64 optimisation dialogs: 

<img width="577" alt="Screenshot 2024-06-10 at 12 38 42" src="https://github.com/Automattic/studio/assets/2998162/f624ae8e-c794-4ab8-ab65-371cd36d7fdb">

<img width="520" alt="Screenshot 2024-06-10 at 11 41 12" src="https://github.com/Automattic/studio/assets/2998162/36db16e2-5998-4ffd-b5e7-da5be2ecd40b">

Wdyt?

Reply:
@matt-west, I'd be interested to hear your perspective on this, too. The three options feel a little cluttered to me:

<img width="563" alt="Screenshot 2024-06-10 at 12 46 40" src="https://github.com/Automattic/studio/assets/2998162/189418e5-0b6a-48b3-b409-86387b9efc9b">

Reply:
> When looking at the dialog, I think it would feel more intuitive for me to see the `Sounds good, do it for me` option at the top of the list, not the bottom.

I originally put the buttons in that order but changed them at the last moment because the default response when canceling the dialog is associated with the first element. I thought this order might be following design guidelines, but I've just checked [the documentation](https://learn.microsoft.com/en-us/windows/apps/design/controls/dialogs-and-flyouts/dialogs) and it's recommended to have the "Yes" action in the first position:

> The "do it" action button(s) should appear as the leftmost buttons. The safe, nondestructive action should appear as the rightmost button.

I'll apply this change.

> I also wondered if it'd make sense for the `I'm not interested` option to follow the same `Don't ask again` pattern we use for the demo update and arm64 optimisation dialogs:

This dialog is only prompted automatically once, so I think we wouldn't need to provide a turn-off option. WDYT?

Reply:
> This dialog is only prompted automatically once, so I think we wouldn't need to provide a turn-off option. WDYT?

Good point, the main motivation behind the suggestion was making `I'm not interested` less prominent, which I think moving to the last position in the button list would accomplish. 👍

Reply:
I've reordered the buttons in https://github.com/Automattic/studio/pull/211/commits/fe04881febd59b507489c704874ff9e66e6a8fbe and https://github.com/Automattic/studio/pull/211/commits/2bf5df494687454a9f92c2aefef34d5296b20b3f.

Reply:
I agree that the "do it for me" option should be first.

Do we need to include the option for manually applying the change? I think we could simplify this dialog by removing that option.

Reply:
> Do we need to include the option for manually applying the change? I think we could simplify this dialog by removing that option.

My rationale behind adding the option was to provide a middle solution where the user could check the documentation. However, if we foresee that the option won't be helpful the user and reduces clarity of the dialog, I'll be happy to remove it. WDYT @matt-west ?

Reply:
Okay. Let's be opinionated here and remove that option. Well need to adjust the body copy to reflect this change too.

Reply:
>  Well need to adjust the body copy to reflect this change too.

Originally, I thought of adding a link to the documentation in the body but the native dialog just renders plain text 😞. Do you think we should reference the documentation or simply offering the option to run the action automatically?

Reply:
I don't think we need to reference the docs here. If we find that users are confused about what Windows Defender is and how it works, we can add some additional copy to the dialog.

Reply:
I applied this suggestion in https://github.com/Automattic/studio/pull/211/commits/f260f97190dc61703653600588ba235fcf73374b.